{
  boolean fail=false;
  extraInfo=extraInfo == null || extraInfo.isEmpty() ? "" : extraInfo + ": ";
  extraInfo+="expected a " + exceptionClass + " exception to be thrown";
  if (status != null) {
    extraInfo+=" with status [" + status + "]";
  }
  try {
    future.actionGet();
    fail=true;
  }
 catch (  ElasticsearchException esException) {
    assertThat(extraInfo,esException.unwrapCause(),instanceOf(exceptionClass));
    if (status != null) {
      assertThat(extraInfo,ExceptionsHelper.status(esException),equalTo(status));
    }
  }
catch (  Exception e) {
    assertThat(extraInfo,e,instanceOf(exceptionClass));
    if (status != null) {
      assertThat(extraInfo,ExceptionsHelper.status(e),equalTo(status));
    }
  }
  if (fail) {
    throw new AssertionError(extraInfo);
  }
}
